src/lib.rs

```diff
@@ -85,27 +86,27 @@ impl<F: Field, PC: PolynomialCommitment<F>> SemanticallyValid for BatchLCProof<F
 /// a sender to commit to multiple polynomials and later provide a succinct proof
 /// of evaluation for the corresponding commitments at a query set `Q`, while
 /// enforcing per-polynomial degree bounds.
-pub trait PolynomialCommitment<F: Field>: Sized {
+pub trait PolynomialCommitment<G: AffineCurve>: Sized {
     /// The universal parameters for the commitment scheme. These are "trimmed"
     /// down to `Self::CommitterKey` and `Self::VerifierKey` by `Self::trim`.
-    type UniversalParams: PCUniversalParams;
+    type Parameters: PCParameters<
+        CommitterKey = Self::CommitterKey,
+        VerifierKey = Self::VerifierKey,
+        Error = Self::Error
+    >;
     /// The committer key for the scheme; used to commit to a polynomial and then
     /// open the commitment to produce an evaluation proof.
     type CommitterKey: PCCommitterKey;
     /// The verifier key for the scheme; used to check an evaluation proof.
     type VerifierKey: PCVerifierKey;
-    /// The prepared verifier key for the scheme; used to check an evaluation proof.
-    type PreparedVerifierKey: PCPreparedVerifierKey<Self::VerifierKey> + Clone;
     /// The commitment to a polynomial.
     type Commitment: PCCommitment + Default + Debug + Eq + PartialEq;
-    /// The prepared commitment to a polynomial.
-    type PreparedCommitment: PCPreparedCommitment<Self::Commitment>;
     /// The commitment randomness.
     type Randomness: PCRandomness;
```

Comment:
There is no need for keeping with the associated type ``Randomness``. We now have defined the Polynomial commitment over an affine curve, so Randomness will always be from the scalar field of it.

Reply:
In domain extended ipa_pc scheme randomness is a vector of scalars and to a single scalar, so PCRandomness trait still needed to give us and abstraction layer